### ISSUE TITLE
Allow trailing Haddock comments

### ISSUE THREAD

**Issue by brandon-leapyear** [parsonsmatt/fourmolu#78]
This is mentioned in https://github.com/parsonsmatt/fourmolu/issues/32#issuecomment-702474279, but I figured it's worth breaking out into a separate issue.

It would be great to configure trailing Haddock comments, specifically for data types. Something like
```hs
data Foo
  = Foo1 -- ^ Foo1
  | Foo2 -- ^ Foo2
```
or even
```hs
data Foo
  = Foo1
    -- ^ Foo1
  | Foo2
    -- ^ Foo2
```

**Comment by georgefst** [comment]
I haven't got time to go looking for the thread right now, but I *think* this was rejected for Ormolu because it would be inconsistent with function arguments. This is currently a parse error, unfortunately:

```hs

f ::
    Int ->
    -- ^ aaa
    Int ->
    -- ^ bbb
    Int
f = undefined
```

This is [fixed](https://downloads.haskell.org/ghc/9.0.1/docs/html/users_guide/9.0.1-notes.html#haddock) in GHC 9.0. But it'll be a while before we can take advantage of that (even if we're only keeping support for three compiler versions, that's around two years at the current rate).

**Comment by georgefst** [comment]
Actually, it might not be too awful for us to do this conditionally on compiler version, but I'd be a bit anxious about that.
